Improve basics aa2 to implement 1

pinterid opened the pull request:
**Checklist:**

 <!--- Make sure your PR is documented and tested before submission. Put an `x` in all the boxes that apply: -->
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?


**Type of Changes:**

 <!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] New feature (non-breaking change which adds functionality)
 - [x] Breaking change (fix or feature that would cause existing functionality to change)

 - [ ] Documentation (non-breaking change which adds documentation)

**What is the current behavior? (link to any open issues here)**
- Currently all logic is located in the `App.js` which produces a lot of mess. #64 
- Currently routing does not work well on static file servers. #66 


**What is the new behavior (if this is a feature change)?**
- Now logic is split into separate actions. `App.js` only fulfils the purpose of setting main states. #64 
- Now routing works even on static file servers.  #66


**Other information**:
- Ref: 🐍

pinterid committed:
Add ghpages workaround

The workaround for ghpages was implemented, it is now possible to reach sub-directories.

pinterid committed:
Add core

The core for actions was added.

pinterid committed:
Add authentication actions

The authentication actions were added.

pinterid committed:
Add general actions

The general actions were added.

pinterid committed:
Add user actions

The user actions were added.

pinterid committed:
Add talk actions

The talks actions were added.

pinterid committed:
Add controlling actions

The controlling actions were added.

pinterid committed:
Add login handler

The login handler was implemented.

pinterid committed:
Add logout handler

The handler for logout was added.

pinterid committed:
Add registration handler

The handler for registrations was added.

pinterid committed:
Add cache renewal handler

The handler for cache renewal was added.

pinterid committed:
Add profile fetching handler

The handler for fetching profiles was added.

pinterid committed:
Add talk upload handler

The handler for talk uploads was added.

pinterid committed:
Add talk deletion handler

The handler for deleting talks was added.

pinterid committed:
Add save settings handler

The handler for saving settings was added.

pinterid committed:
Add control handler

The handler for controlling data were added.

pinterid committed:
Add auto login

Automatically login as a real or anonymous user on start.

pinterid committed:
Update function access

The function access was updated to the new structure.

pinterid committed:
Update imports

An unnecessary import was removed.

pinterid committed:
Fix talk tab

The wrong using of links in talks was removed.

pinterid committed:
Add themes controlling

The site no longer crashes if a user has no themes.

pinterid committed:
Update upload modal

The talk upload modal structure was improved.

pinterid committed:
Fix login fail error

The login fail error was patched.

pinterid committed:
Improve code quality

The code quality was improved due to a request by @kleberbaum.

pinterid committed:
Update navbar link

The navbar links were updated from regular html to MDB Link.

Aichnerc commented on public/404.html:
```diff
@@ -0,0 +1,40 @@
+<html>
+  <head>
+    <meta charset="utf-8">
+    <title>Single Page Apps for GitHub Pages</title>
```
I guess that's not a good name for snek.

This name will apear in meta information when a link is posted in a messenger, facebook, linkedin, ... or any other form that fetches meta data.

The favicon and all other meta tags have to be added to this file as well.

Aichnerc commented on public/404.html:
```diff
@@ -0,0 +1,40 @@
+      // https://username.github.io/repo-name/one/two?a=b&c=d#qwe becomes
+      // https://username.github.io/repo-name/?p=/one/two&q=a=b~and~c=d#qwe
+      // Otherwise, leave segmentCount as 0.
+      var segmentCount = 0;
```
https://en.wikipedia.org/wiki/Hanging

schettn replied:
![image](https://user-images.githubusercontent.com/52858351/85026360-de2e7580-b178-11ea-9a74-06b1da5341bf.png)

Ref: https://github.com/aichner/Gutschein2GoShopify/blob/994955910453758155cb92e87c9397953ca75dc5/utils.js

Mhmmm... xD

schettn replied:
Listen to my new rule. Never fucking ever touch the deployment routing workaround files again!!!1!!!!1

Aichnerc commented on public/404.html:
```diff
@@ -0,0 +1,40 @@
+      // Otherwise, leave segmentCount as 0.
+      var segmentCount = 0;
+
+      var l = window.location;
```
https://en.wikipedia.org/wiki/Hanging

schettn replied:
https://github.com/snek-at/front/pull/65/files#r445154486

Aichnerc commented on public/index.html:
```diff
@@ -10,23 +10,55 @@
+      // the single page app to route accordingly.
+      (function(l) {
+        if (l.search) {
+          var q = {};
```
https://en.wikipedia.org/wiki/Hanging

schettn replied:
https://github.com/snek-at/front/pull/65/files#r445154486

Aichnerc commented on public/index.html:
```diff
@@ -10,23 +10,55 @@
+        if (l.search) {
+          var q = {};
+          l.search.slice(1).split('&').forEach(function(v) {
+            var a = v.split('=');
```
https://en.wikipedia.org/wiki/Hanging

schettn replied:
https://github.com/snek-at/front/pull/65/files#r445154486

Aichnerc commented on src/App.js:
```diff
@@ -23,6 +21,25 @@ import { Footer, Navbar } from "./components/molecules";
+  getTalk,
+  uploadTalk,
+  deleteTalk,
+} from "./actions";
```
I would split this up in mutliple categories.

For instance, you can create a group for talks. This eases the mind and improve readability.

schettn replied:
Yep. I agree. I suggest following structure:

**For all action sections:**

```
import {
  getAllTalks,
  getTalk,
  uploadTalk,
  deleteTalk,
} from "actions/intel/talksActions";
```

**For the action core:**

`import { intelFerry } from "actions/core"`

- As you can see I changed `ferry` to `intelFerry` in order to prevent naming conflicts for future endpoints like "intel".
- Rename `intel` to `intel`-> every front folder is lowercase
- Adjust the import structure, as mentioned by @Aichnerc.

pinterid replied:
Can we have a meeting about this before changing the structure?

Aichnerc commented on src/App.js:
```diff
@@ -37,551 +54,226 @@ class App extends React.Component {
-      return false;
-    }
+  globalFunctions = {
+    /** Authentication Actions*/
```
Since when do we use this type of commenting in the front?

Why not use `//`?

schettn replied:
I'm sorry. Might be my influence ^^ but not wanted in a non typescript file.

Aichnerc commented on src/App.js:
```diff
@@ -37,551 +54,226 @@ class App extends React.Component {
-  usernameMatchesFetchedUsername = (username) => {
-    return username === this.state.fetchedUser?.platformData?.profile?.username;
+  componentDidMount = () => {
+    this.handleLoginSession();
```
Please descripe what this does.

Aichnerc commented on src/actions/Intel/authActions.js:
```diff
@@ -0,0 +1,65 @@
+          return {
+            username: whoami.username,
+            avatarUrl:
+              "https://www.clipartmax.com/png/full/166-1669056_the-20-cooler-octocat-github-octocat.png",
```
It is wise to use such a big picture?

What about this picture? It gives me less anxiety than the other one.

<img src="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcQT9FAXb9EMeULwwzTJ7ch7Ckx4qq6x1WMIilOVjaAtXSMKGlS8&usqp=CAU" height="200" />

32x32px should be enough.

schettn replied:
> It is wise to use such a big picture?
> 
> What about this picture? It gives me less anxiety than the other one.
> 
> <img alt="" height="200" src="https://camo.githubusercontent.com/7be1ef51e492043184abe241cdd8e6b2dc9b3af8/68747470733a2f2f656e637279707465642d74626e302e677374617469632e636f6d2f696d616765733f713d74626e253341414e643947635154394641586239454d65554c77777a544a37636837436b78347171367831574d49696c4f566a61417458534d4b476c533826757371703d434155">
> 
> 32x32px should be enough.

Looks nice xD. We should take this one.

pinterid replied:
Change will be implement if you provide me a link that is shorter than:
"https://camo.githubusercontent.com/7be1ef51e492043184abe241cdd8e6b2dc9b3af8/68747470733a2f2f656e637279707465642d74626e302e677374617469632e636f6d2f696d616765733f713d74626e253341414e643947635154394641586239454d65554c77777a544a37636837436b78347171367831574d49696c4f566a61417458534d4b476c533826757371703d434155"

simonprast replied:
32x32 is too small. Use 256x256 jpeg for now per favor. @Aichnerc's octocat is 225x225.

Aichnerc commented on src/actions/Intel/generalActions.js:
```diff
@@ -0,0 +1,175 @@
+      return {
+        errorCode: 2,
+        message: "Appending source objects failed",
+        raw: ex,
```
rawr.

<img src="https://user-images.githubusercontent.com/21159914/84789386-76e5b980-aff0-11ea-92a0-9833b49a102d.png" alt="Furrytales" height="250"/>

<sub><sup>Disclaimer: I do not approve the existance of furries</sup></sub>

schettn replied:
![image](https://user-images.githubusercontent.com/52858351/85020988-7163ad00-b171-11ea-9d78-272562879edf.png)

pinterid replied:
Does anything has to be changes at this code part??

schettn replied:
nep

Aichnerc replied:
Best thread on this platform. ☠

Aichnerc reviewed (changes requested):
I've seen some things.

<sup><sub>Terrible things.</sup></sub>

schettn reviewed (changes requested):
I would improve the following:

**- Profile Page Errors**
When using the search to navigate to a user, everything works like a charm but navigating to routes by writing the URL directly, this error occurs. It seems like the workaround does not workaround.
![image](https://user-images.githubusercontent.com/52858351/85025701-049fe100-b178-11ea-9a7d-3bf9f6b57467.png)

**- https://github.com/snek-at/front/pull/65#discussion_r442189240**
**- There are duplicates in the error codes**
We should have unique error codes and add a description to them in future. When we do not add good documentation for the error codes we end like https://github.com/amzn.. Spoiler: (Their error code documentation sucks balls)
![image](https://user-images.githubusercontent.com/52858351/85023510-17fd7d00-b175-11ea-8bc4-69cafc600b7a.png)

 
**- Documentation of actions**
In my opinion, the comments of the actions and other code are just garbage that does not serve the purpose of providing information. The comment below is a prime example of this problem.


Documentation Example:
![image](https://user-images.githubusercontent.com/52858351/85021797-a6bcca80-b172-11ea-8f49-435d432fabc3.png)

- What should `A data object to store in cache` testify? Btw.. wtf.. What is a data object?
- The description is not enough either. `Handles the calls for writing to cache`? Where is the data written to?
- Cache? What kind of cache local cache, or something backend related?

**I would suggest following steps:**

- [x] Fix routing bug on deployment
- [x] Resolve https://github.com/snek-at/front/pull/65#discussion_r442189240
- [x] Make error codes unique
- [x] Hold a meeting regarding documentation. Especially JSDocs Headers 
   - [ ] Improve the documentation of actions and App.js

pinterid committed:
Improve code quality

The code quality was improved  due to a request by @Aichnerc.

pinterid committed:
Improve structure

The structure was improved due to a request by @Aichnerc and a discussion with @schettn & @kleberbaum.

schettn committed:
Some minor structure changes in actions

Intel is written in lowercase now.
Duplciate // are removed now

schettn committed:
Move username to first level fetchUser 

Now the username is not longer notated under `fetchUser.plartformData.profile.username` instead under
`fetchUser.username`.

schettn committed:
Fix talks page rendering

Because of a stupid allocation the talks page rendering
failed. I repaired it and now it works.. :)

schettn committed:
Rework error codes

There were duplicates in error codes which are resolved now.
In addition all error codes are now in the range of 600 - 699.
SNEK codes.

pinterid committed:
Add oauth utils

The utilites required for OAuth were added.

pinterid committed:
Add window opener

The window opener required for OAuth was added.

pinterid committed:
Add oauth component

The oauth component was added.

pinterid committed:
Update import/exports 

The import & exports required for OAuth were updated.

pinterid committed:
Implement oauth component

The oauth component was implemented.

pinterid committed:
Update routes

The routes required for oauth were updated.

pinterid committed:
Update package.json

The package.json was updated, it is now possible to deploy the project.

schettn committed:
Just to save it

routing sucks..

schettn committed:
Finally fixed the fucking routing issues

And here we are. After three days of shit the routing issue is
resolved and everything works like a charm.
For the nerds:
THE ISSUE WAS THE FUCKING HOMEPAGE PATH IN THE
PACKAGE.JSON.. HOW THE FUCK COULD THIS HAPPEN???

schettn committed:
Lowerize Intel Folder Name

Intel is changed to intel now.

pinterid committed:
Update GitHub OAuth component

The github oauth component is now received from a npm package.

schettn commented on package.json:
```diff
@@ -2,10 +2,12 @@
   "dependencies": {
     "fuzzysort": "^1.1.4",
+    "gh-pages": "^3.1.0",
+    "ghspa": "^1.0.0",
```
Whats that?

schettn commented on public/404.html:
```diff
@@ -0,0 +1,40 @@
+      var segmentCount = 0;
+
+      var l = window.location;
+      console.log(window.location, "aaaaa");
```
I know I said that nobody should touch this file again but... the console.log is stupid and should be removed. ^^

schettn commented on public/404.html:
```diff
@@ -0,0 +1,40 @@
+
+    </script>
+  </head>
+</html>
```
Add newline

schettn commented on public/CNAME:
```diff
@@ -0,0 +1 @@
+snek.at
```
Does a newline work in this file? If so then add it.

schettn commented on public/fav/manifest.json:
```diff
@@ -0,0 +1,41 @@
+   "density": "4.0"
+  }
+ ]
+}
```
Newline?

Aichnerc reviewed (approved):
♋

schettn commented on src/App.js:
```diff
@@ -37,588 +55,247 @@ class App extends React.Component {
+    // Controlling Actions
+    refetchRequired: (username) => this.refetchRequired(username),
+    usernameMatchesFetchedUsername: (username) =>
+      this.usernameMatchesFetchedUsername(username),
```
```suggestion
    // Authentication
    login: async (username, password) =>
      this.handleLoginSession({ username, password: sha256(password) }),
    logout: async () => this.handleLogout(),
    // General
    fetchGitLabServers: async () => ferry(fetchGitLabServers()),
    appendSourceObjects: async (sourceList) =>
      ferry(appendSourceObjects(sourceList)),
    users: async () => ferry(getAllPageUrls()),
    saveSettings: async (nextSettings) => this.handleSaveSettings(nextSettings),
    // User
    updateCache: async (fetchedUser) => this.handleCacheRenewal(fetchedUser),
    writeCache: async (platformData) => ferry(writeCache(platformData)),
    registerUser: async (registrationData) =>
      this.handleRegistration(registrationData),
    fetchCacheData: async (username) => this.handleProfileFetching(username),
    // Talk Actions
    deleteTalk: async (talk) => this.handleTalkDeletion(talk),
    uploadTalk: async (file, talkInfo) => this.handleTalkUpload(file, talkInfo),
    getTalk: (uid, username) => ferry(getTalk(uid, username)),
    // State checking
    refetchRequired: (username) => this.refetchRequired(username),
    usernameMatchesFetchedUsername: (username) =>
      this.usernameMatchesFetchedUsername(username),
```

schettn commented on src/components/molecules/forms/RegisterForm/index.jsx:
```diff
@@ -599,9 +603,25 @@ class RegisterForm extends React.Component {
+                  onSuccess={this.oauthGitHubSuccess}
+                  onFailure={this.oauthGitHubFailure}
+                />
+              )}
```
This is not a god way to check whether its development or not.. 

Please change it to:
```
if (!process.env.NODE_ENV || process.env.NODE_ENV === "development") {
// Development Code
} else {
// Production Code
}
```

schettn reviewed (changes requested):
Look at my suggestions and remove all console logs which are not meant to used for troubleshooting.
For those troubleshooting logs add a TSID anchor and so on..

simonprast commented on src/actions/core.js:
```diff
@@ -0,0 +1,41 @@
+
+/**
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © 2019-2020 Simon Prast
```
Remove the year number for consitency

simonprast commented on src/actions/index.js:
```diff
@@ -0,0 +1,14 @@
+
+/**
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © 2019-2020 Simon Prast
```
Remove the year number for consitency

schettn replied:
I will add the year to the missing tags for consistency.

simonprast commented on src/actions/intel/authActions.js:
```diff
@@ -0,0 +1,65 @@
+
+/**
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © 2019-2020 Simon Prast
```
Remove the year number for consitency

simonprast commented on src/App.js:
```diff
@@ -23,6 +21,26 @@ import { Footer, Navbar } from "./components/molecules";
 //> Routes
 import Routes from "./Routes";
+//> Core
+import { ferry } from "./actions";
```
Could this be given a meaningful name or could be described what it's for?

schettn replied:
I will rename it to intelFerry due to future ferry implementations.

A description is already added inside `./actions`

simonprast commented on src/actions/intel/generalActions.js:
```diff
@@ -0,0 +1,175 @@
+
+/**
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © 2019-2020 Simon Prast
```
Remove the year number for consitency

simonprast commented on src/actions/intel/index.js:
```diff
@@ -0,0 +1,40 @@
+
+/**
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © 2019-2020 Simon Prast
```
Remove the year number for consitency

simonprast commented on src/actions/intel/talksActions.js:
```diff
@@ -0,0 +1,133 @@
+
+/**
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © 2019-2020 Simon Prast
```
Remove the year number for consitency

simonprast commented on src/actions/intel/userActions.js:
```diff
@@ -0,0 +1,250 @@
+
+/**
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © 2019-2020 Simon Prast
```
Remove the year number for consitency

simonprast reviewed (changes requested):
appearance stuff

schettn committed:
Improve code quality

The code quality has been improved due to a request of @schettn

schettn committed:
Rename ferry to intelFerry

The ferry has been renamed due to future implementations of
ferry.

schettn committed:
Refined console.logs

Useless logs are removed now. Some logs are changed to errors
now.

schettn committed:
Adjust licence tags

The year was missing in some licence tags

schettn reviewed (approved):
Looks good.

simonprast reviewed (approved):
Okay nice

schettn committed:
Merge branch 'implement-1' into improve-basics-aa2
